### DISCUSSION TITLE
Animation Nodes Centos 7

### DISCUSSION THREAD

**Discussion by GiorgioAdolfo** [JacquesLucke/animation_nodes#1692] [General]
Hello all!
I have been trying to compile AN on Centos 7 which is now officially supported by Blender, but keep getting errors. It would be really nice if we could get an official AN version with support for Centos 7. 

Thank you all!!
Cheers!

**Comment by OmarEmaraDev** [comment]
What are the errors exactly?

**Reply by GiorgioAdolfo** [reply]
Hello..
Here are error message:
Traceback (most recent call last): File "/opt/Autodesk/flame_2021/blender-2.83-839fe335d2d6-linux64/2.83/scripts/modules/addon_utils.py", line 351, in enable mod = import(module_name) File "/opt/Autodesk/flame_2021/.config/blender/2.83/scripts/addons/animation_nodes/init.py", line 166, in <module> auto_load.init() File "/opt/Autodesk/flame_2021/.config/blender/2.83/scripts/addons/animation_nodes/auto_load.py", line 23, in init modules = get_all_submodules(Path(file).parent) File "/opt/Autodesk/flame_2021/.config/blender/2.83/scripts/addons/animation_nodes/auto_load.py", line 51, in get_all_submodules return list(iter_submodules(directory, directory.name)) File "/opt/Autodesk/flame_2021/.config/blender/2.83/scripts/addons/animation_nodes/auto_load.py", line 55, in iter_submodules yield importlib.import_module("." + name, package_name) File "/opt/Autodesk/flame_2021/blender-2.83-839fe335d2d6-linux64/2.83/python/lib/python3.7/importlib/init.py", line 127, in import_module return _bootstrap._gcd_import(name[level:], package, level) File "/opt/Autodesk/flame_2021/.config/blender/2.83/scripts/addons/animation_nodes/libs/FastNoiseSIMD/init.py", line 1, in <module> from . node_base import Noise3DNodeBase File "/opt/Autodesk/flame_2021/.config/blender/2.83/scripts/addons/animation_nodes/libs/FastNoiseSIMD/node_base.py", line 6, in <module> from . wrapper import ( ImportError: /opt/Autodesk/flame_2021/.config/blender/2.83/scripts/addons/animation_nodes/libs/FastNoiseSIMD/wrapper.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _ZN22FastNoiseSIMD_internal16FastNoiseSIMD_L3C1Ei</module></module></module>

**Reply by GiorgioAdolfo** [reply]
Those errors are when i tried to load the plugin inside blender.

**Reply by GiorgioAdolfo** [reply]
Just tried to compile again, and noticed it is something related to FASTNOISESimd :
Compile Libraries
--------------------------------------------------

Compile FastNoiseSIMD

In file included from FastNoiseSIMD_avx2.cpp:48:0:
FastNoiseSIMD_internal.cpp: In function ‘SIMDf L3_FUNC_CellularValueEuclideanSingle(SIMDi, SIMDf, SIMDf, SIMDf, SIMDf)’:
FastNoiseSIMD_internal.cpp:293:47: error: ‘_mm256_undefined_ps’ was not declared in this scope
 #define SIMDf_UNDEFINED() _mm256_undefined_ps()
                                               ^
FastNoiseSIMD_internal.cpp:1581:20: note: in expansion of macro ‘SIMDf_UNDEFINED’
  SIMDf cellValue = SIMDf_UNDEFINED();\
                    ^
FastNoiseSIMD_internal.cpp:1903:1: note: in expansion of macro ‘CELLULAR_VALUE_SINGLE’
 CELLULAR_VALUE_SINGLE(Euclidean)
 ^
FastNoiseSIMD_internal.cpp: In function ‘SIMDf L3_FUNC_CellularValueManhattanSingle(SIMDi, SIMDf, SIMDf, SIMDf, SIMDf)’:
FastNoiseSIMD_internal.cpp:293:47: error: ‘_mm256_undefined_ps’ was not declared in this scope
 #define SIMDf_UNDEFINED() _mm256_undefined_ps()
                                               ^
FastNoiseSIMD_internal.cpp:1581:20: note: in expansion of macro ‘SIMDf_UNDEFINED’
  SIMDf cellValue = SIMDf_UNDEFINED();\
                    ^
FastNoiseSIMD_internal.cpp:1904:1: note: in expansion of macro ‘CELLULAR_VALUE_SINGLE’
 CELLULAR_VALUE_SINGLE(Manhattan)
 ^
FastNoiseSIMD_internal.cpp: In function ‘SIMDf L3_FUNC_CellularValueNaturalSingle(SIMDi, SIMDf, SIMDf, SIMDf, SIMDf)’:
FastNoiseSIMD_internal.cpp:293:47: error: ‘_mm256_undefined_ps’ was not declared in this scope
 #define SIMDf_UNDEFINED() _mm256_undefined_ps()
                                               ^
FastNoiseSIMD_internal.cpp:1581:20: note: in expansion of macro ‘SIMDf_UNDEFINED’
  SIMDf cellValue = SIMDf_UNDEFINED();\
                    ^
FastNoiseSIMD_internal.cpp:1905:1: note: in expansion of macro ‘CELLULAR_VALUE_SINGLE’
 CELLULAR_VALUE_SINGLE(Natural)
 ^
FastNoiseSIMD_internal.cpp: In function ‘SIMDf L3_FUNC_CellularLookupEuclideanSingle(SIMDi, SIMDf, SIMDf, SIMDf, SIMDf, const NoiseLookupSettings&)’:
FastNoiseSIMD_internal.cpp:293:47: error: ‘_mm256_undefined_ps’ was not declared in this scope
 #define SIMDf_UNDEFINED() _mm256_undefined_ps()
                                               ^
FastNoiseSIMD_internal.cpp:1670:16: note: in expansion of macro ‘SIMDf_UNDEFINED’
  SIMDf xCell = SIMDf_UNDEFINED();\
                ^
FastNoiseSIMD_internal.cpp:1907:1: note: in expansion of macro ‘CELLULAR_LOOKUP_SINGLE’
 CELLULAR_LOOKUP_SINGLE(Euclidean)
 ^
FastNoiseSIMD_internal.cpp: In function ‘SIMDf L3_FUNC_CellularLookupManhattanSingle(SIMDi, SIMDf, SIMDf, SIMDf, SIMDf, const NoiseLookupSettings&)’:
FastNoiseSIMD_internal.cpp:293:47: error: ‘_mm256_undefined_ps’ was not declared in this scope
 #define SIMDf_UNDEFINED() _mm256_undefined_ps()
                                               ^
FastNoiseSIMD_internal.cpp:1670:16: note: in expansion of macro ‘SIMDf_UNDEFINED’
  SIMDf xCell = SIMDf_UNDEFINED();\
                ^
FastNoiseSIMD_internal.cpp:1908:1: note: in expansion of macro ‘CELLULAR_LOOKUP_SINGLE’
 CELLULAR_LOOKUP_SINGLE(Manhattan)
 ^
FastNoiseSIMD_internal.cpp: In function ‘SIMDf L3_FUNC_CellularLookupNaturalSingle(SIMDi, SIMDf, SIMDf, SIMDf, SIMDf, const NoiseLookupSettings&)’:
FastNoiseSIMD_internal.cpp:293:47: error: ‘_mm256_undefined_ps’ was not declared in this scope
 #define SIMDf_UNDEFINED() _mm256_undefined_ps()
                                               ^
FastNoiseSIMD_internal.cpp:1670:16: note: in expansion of macro ‘SIMDf_UNDEFINED’
  SIMDf xCell = SIMDf_UNDEFINED();\
                ^
FastNoiseSIMD_internal.cpp:1909:1: note: in expansion of macro ‘CELLULAR_LOOKUP_SINGLE’
 CELLULAR_LOOKUP_SINGLE(Natural)
 ^
Done.


Compile

**Reply by OmarEmaraDev** [reply]
@GiorgioAdolfo Thanks! Which compiler is this? And what is the version of the toolchain?

**Reply by GiorgioAdolfo** [reply]
Thanks!! Here are the list of the compilers installed on the system:
gcc.x86_64                                4.8.5-44.el7               @base      
gcc-c++.x86_64                            4.8.5-44.el7               @base      
gcc-gfortran.x86_64                       4.8.5-44.el7               @base      
libgcc.i686                               4.8.5-44.el7               @base      
libgcc.x86_64                             4.8.5-44.el7               @base

**Reply by GiorgioAdolfo** [reply]
java-1.6.0-openjdk.x86_64                 1:1.6.0.41-1.13.13.1.el7_3 @anaconda  
java-1.6.0-openjdk-devel.x86_64           1:1.6.0.41-1.13.13.1.el7_3 @anaconda  
java-1.7.0-openjdk.x86_64                 1:1.7.0.191-2.6.15.5.el7   @anaconda  
java-1.7.0-openjdk-devel.x86_64           1:1.7.0.191-2.6.15.5.el7   @anaconda  
java-1.7.0-openjdk-headless.x86_64        1:1.7.0.191-2.6.15.5.el7   @anaconda  
java-1.8.0-openjdk.x86_64                 1:1.8.0.181-7.b13.el7      @anaconda  
java-1.8.0-openjdk-devel.x86_64           1:1.8.0.181-7.b13.el7      @anaconda  
java-1.8.0-openjdk-headless.x86_64        1:1.8.0.181-7.b13.el7      @anaconda  
javamail.noarch                           1.4.6-8.el7                @anaconda  
javapackages-tools.noarch                 3.4.1-11.el7               @anaconda  
javassist.noarch                          3.16.1-10.el7              @anaconda  
libobasis7.0-extension-javascript-script-provider.x86_64
libvirt-java.noarch                       0.4.9-4.el7                @anaconda  
libvirt-java-devel.noarch                 0.4.9-4.el7                @anaconda  
openoffice-javafilter.x86_64              4.1.1-9775                 installed  
python-javapackages.noarch                3.4.1-11.el7               @anaconda  
tzdata-java.noarch                        2018e-3.el7                @anaconda

**Reply by GiorgioAdolfo** [reply]
Group: Development Tools
 Group-Id: development
 Description: A basic development environment.
 Mandatory Packages:
    autoconf
    automake
    binutils
    bison
    flex
    gcc
    gcc-c++
    gettext
    libtool
    make
    patch
    pkgconfig
    redhat-rpm-config
    rpm-build
    rpm-sign
 Default Packages:
    byacc
    cscope
    ctags
    diffstat
    doxygen
    elfutils
    gcc-gfortran
    git
    indent
    intltool
    patchutils
    rcs
    subversion
    swig
    systemtap
 Optional Packages:
   ElectricFence
   ant
   babel
   bzr
   ccache
   chrpath
   clips
   clips-devel
   clips-doc
   clips-emacs
   clips-xclips
   clipsmm-devel
   clipsmm-doc
   cmake
   cmucl
   colordiff
   compat-gcc-44
   compat-gcc-44-c++
   cvs
   cvsps
   darcs
   dejagnu
   email2trac
   expect
   ftnchek
   gcc-gnat
   gcc-objc
   gcc-objc++
   ghc
   git
   haskell-platform
   imake
   javapackages-tools
   ksc
   lua
   mercurial
   mock
   mod_dav_svn
   nasm
   nqc
   nqc-doc
   ocaml
   perltidy
   qgit
   rpmdevtools
   rpmlint
   sbcl
   scorep
   systemtap-sdt-devel
   systemtap-server
   trac
   trac-git-plugin
   trac-mercurial-plugin
   trac-webadmin
   translate-toolkit

**Reply by OmarEmaraDev** [reply]
@GiorgioAdolfo It seems GCC 4.8.5 is too old to be used to compile Animation Nodes. You need a more recent compiler.

**Reply by GiorgioAdolfo** [reply]
Thanks!! That was the answer, installed a new version of GCC, and the ToolChain. I was able to compile and install AN.
Will leave the links here, so others can try to see if it works.
Devel Tools:
https://www.softwarecollections.org/en/scls/rhscl/devtoolset-8/

Installing GCC
https://gist.github.com/nchaigne/ad06bc867f911a3c0d32939f1e930a11
